### PR TITLE
CI: Pass environment variables directly to configure in JIT CI

### PR DESCRIPTION
**Pull request by erlend-aasland**

### CODE REVIEW

File: .github/workflows/jit.yml [hunk 1]
```diff
@@ -96,7 +96,7 @@ jobs:
         if: runner.os == 'macOS'
         run: |
           brew install llvm@${{ matrix.llvm }}
-          export SDKROOT="$(xcrun --show-sdk-path)"
+          SDKROOT="$(xcrun --show-sdk-path)" \
           ./configure --enable-experimental-jit ${{ matrix.debug && '--with-pydebug' || '--enable-optimizations --with-lto' }}
```
Comment (hugovk) [inline .github/workflows/jit.yml]: Add an indent to make it clearer the line is being continued?
```suggestion
            ./configure --enable-experimental-jit ${{ matrix.debug && '--with-pydebug' || '--enable-optimizations --with-lto' }}
```

File: .github/workflows/jit.yml [hunk 2]
```diff
@@ -122,10 +122,10 @@ jobs:
           sudo apt install --yes "gcc-$HOST" qemu-user
           ${{ !matrix.debug && matrix.compiler == 'clang' && './configure --enable-optimizations' || '' }}
           ${{ !matrix.debug && matrix.compiler == 'clang' && 'make profile-run-stamp --jobs 4' || '' }}
-          export CC="${{ matrix.compiler == 'clang' && 'clang --target=$HOST' || '$HOST-gcc' }}"
-          export CPP="$CC --preprocess"
-          export HOSTRUNNER=qemu-${{ matrix.architecture }}
           export QEMU_LD_PREFIX="/usr/$HOST"
+          CC="${{ matrix.compiler == 'clang' && 'clang --target=$HOST' || '$HOST-gcc' }}" \
+          CPP="$CC --preprocess" \
+          HOSTRUNNER=qemu-${{ matrix.architecture }} \
```
Comment (hugovk) [inline .github/workflows/jit.yml]:
```suggestion
            CPP="$CC --preprocess" \
            HOSTRUNNER=qemu-${{ matrix.architecture }} \
```

File: .github/workflows/jit.yml [hunk 3]
```diff
@@ -122,10 +122,10 @@ jobs:
...
           ./configure --enable-experimental-jit ${{ matrix.debug && '--with-pydebug' || '--enable-optimizations --with-lto' }} --build=x86_64-linux-gnu --host="$HOST" --with-build-python=../build/bin/python3 --with-pkg-config=no ac_cv_buggy_getaddrinfo=no ac_cv_file__dev_ptc=no ac_cv_file__dev_ptmx=yes
```
Comment (hugovk) [inline .github/workflows/jit.yml]:
```suggestion
            ./configure --enable-experimental-jit ${{ matrix.debug && '--with-pydebug' || '--enable-optimizations --with-lto' }} --build=x86_64-linux-gnu --host="$HOST" --with-build-python=../build/bin/python3 --with-pkg-config=no ac_cv_buggy_getaddrinfo=no ac_cv_file__dev_ptc=no ac_cv_file__dev_ptmx=yes
```

File: .github/workflows/jit.yml [hunk 4]
```diff
@@ -122,10 +122,10 @@ jobs:
...
           ${{ !matrix.debug && matrix.compiler == 'clang' && 'make profile-run-stamp --jobs 4' || '' }}
-          export CC="${{ matrix.compiler == 'clang' && 'clang --target=$HOST' || '$HOST-gcc' }}"
-          export CPP="$CC --preprocess"
-          export HOSTRUNNER=qemu-${{ matrix.architecture }}
           export QEMU_LD_PREFIX="/usr/$HOST"
-          ./configure --enable-experimental-jit ${{ matrix.debug && '--with-pydebug' || '--enable-optimizations --with-lto' }} --build=x86_64-linux-gnu --host="$HOST" --with-build-python=../build/bin/python3 --with-pkg-config=no ac_cv_buggy_getaddrinfo=no ac_cv_file__dev_ptc=no ac_cv_file__dev_ptmx=yes
+          CC="${{ matrix.compiler == 'clang' && 'clang --target=$HOST' || '$HOST-gcc' }}" \
+            CPP="$CC --preprocess" \
+            HOSTRUNNER=qemu-${{ matrix.architecture }} \
+            ./configure --enable-experimental-jit ${{ matrix.debug && '--with-pydebug' || '--enable-optimizations --with-lto' }} --build=x86_64-linux-gnu --host="$HOST" --with-build-python=../build/bin/python3 --with-pkg-config=no ac_cv_buggy_getaddrinfo=no ac_cv_file__dev_ptc=no ac_cv_file__dev_ptmx=yes
```
Comment (hugovk) [inline .github/workflows/jit.yml:129]: This ~300-char line could benefit from splitting with `\` too, but not directly relevant here and not a blocker.

**Review by hugovk — APPROVED** [review]

File: .github/workflows/jit.yml (hunk 4 above)
Comment (erlend-aasland) [reply .github/workflows/jit.yml:129]: Agreed. Suggesting to clean up the whole workflow in a follow-up PR.

**Pull request merged**